packages/reports/addon/components/filter-values/time-dimension/base-interval.ts

```diff
@@ -45,43 +50,39 @@ export default class BaseIntervalComponent extends BaseTimeDimensionFilter {
   /**
    * end of interval
    */
-  @computed('interval', 'calendarDateTimePeriod')
+  @computed('interval', 'grain')
   get endDate() {
-    const { interval, calendarDateTimePeriod } = this;
+    const { interval, grain } = this;
     if (interval) {
-      let { end } = interval.asMomentsForTimePeriod(calendarDateTimePeriod, false);
-      end = end
-        .clone()
-        .subtract(1, getPeriodForGrain(calendarDateTimePeriod))
-        .utc(true);
-      return end;
+      let { end } = interval.asMomentsForTimePeriod(grain);
+      return end.subtract(1, getPeriodForGrain(grain));
```

Comment:
Since we do `end.subtract(...` quite a bit, would it make sense to add that logic to `asMomentsForTimePeriod`?
```js
const inclusiveEnd = true;
let { end } = interval.asMomentsForTimePeriod(grain, inclusiveEnd);
```